nt-app/src/components/vRoomFlags.vue

```diff
@@ -189,40 +195,112 @@ export default {
+            const filePath = dialogReturn.filePaths[0]
 
-                fs.readFile(filePath, 'utf-8', (err, data) => {
-                  if(err){
-                    alert("An error ocurred reading the file :" + err.message);
-                    return;
-                  }
+            fs.readFile(filePath, 'utf-8', (err, data) => {
+              if(err){
+                alert("An error occurred reading the file :" + err.message);
+                return;
+              }
 
-                  // Change how to handle the file content
-                  console.log("The file content is : " + data);
-                  const oldSeed = this.tempFlags.NT_sync_world_seed.value
-                  //TODO potential security issue? Validate this input
-                  this.tempFlags = JSON.parse(data)
-                  this.tempFlags.NT_sync_world_seed.value = oldSeed
-                  this.$refs.orbInput.$refs.input.dispatchEvent(new Event("input"))
-                  this.$refs.seedInput.$refs.input.dispatchEvent(new Event("input"))
-                });
-              });
+              // Change how to handle the file content
+              this.logger.debug(`The file content is : ${data}`)
+              const oldSeed = this.tempFlags.NT_sync_world_seed.value
+              this.loadValidJsonInput(data)
+              this.tempFlags.NT_sync_world_seed.value = oldSeed
+              this.$refs.orbInput.$refs.input.dispatchEvent(new Event("input"))
+              this.$refs.seedInput.$refs.input.dispatchEvent(new Event("input"))
+            });            
+            this.loadedPreset = true
+            this.logger.debug(`loadPresets: ${this.loadedPreset}`)
+          });
+        },
+        loadValidJsonInput(jsonString){
+          let presetData = {}
+          try {
+            presetData = JSON.parse(jsonString)
+          } catch (e) {
+            if (e instanceof SyntaxError) {
+              dialog.showMessageBox({
+                type: "error",
+                title: "Invalid JSON",
+                message: "The file cannot be parsed as valid JSON.",
+                buttons: ["OK"]
+              })
+              return
+            }
+            throw e
+          }
+          const schemaVersion = presetData.schemaVersion || 0
+          this.logger.debug(`Loading preset with schemaVersion: ${schemaVersion}`)
+          switch (schemaVersion) {
+            case 0:
+              this.tempFlagsFromJson(presetData)
+              break;
+            case 1.0:
+              this.tempFlagsFromJson(presetData.gameFlags)
+              this.tempModFlags = new Map();
+              this.tempModFlags = new Map(Object.entries(presetData.modFlags || {}))
+          }
+          this.logger.debug(`tempModFlags loaded from preset file: ${JSON.stringify(Object.fromEntries(this.tempModFlags))}`)
+        },
+        tempFlagsFromJson(inputFlags){
+          let invalidFlags = {}
+          for (const key in inputFlags) {
```

Comment:
```suggestion
          for (const key of Object.keys(inputFlags)) {
```

`for in` is not recommended in JS since it enumerates prototype properties as well